library/Vanilla/Embeds/SoundCloudEmbed.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: chris
+ * Date: 2018-06-05
+ * Time: 5:55 PM
+ */
+
+namespace Vanilla\Embeds;
+
+
+class SoundCloudEmbed extends Embed {
+
+    /** @inheritdoc */
+    protected $type = 'soundcloud';
+
+    /** @inheritdoc */
+    protected $domains = ['soundcloud.com'];
+
+    public function __construct() {
+        parent::__construct('soundcloud', 'image');
+    }
+
+    /**
+     * @inheritdoc
+     *
+     */
+    function matchUrl(string $url) {
+        $data = [];
+        $oembedData =[];
+        $encodedUrl = urlencode($url);
+
+        if ($this->isNetworkEnabled()) {
+            $oembedData = $this->oembed("https://soundcloud.com/oembed?url=" . $encodedUrl . "&format=json");
+            if (array_key_exists('html', $oembedData)) {
+                $data = $this->parseResponseHtml($oembedData['html']);
+            }
+        }
+
+        $data['attributes']['url'] = htmlspecialchars("https://w.soundcloud.com/player/?url=https://api.soundcloud.com/tracks/");
```

Comment:
If this is going to be a static value, is it necessary to save it with every embed?